libbeat/management/management.go

```diff
@@ -91,13 +120,32 @@ func defaultModeConfig() *modeConfig {
 }
 
 // nilManager, fallback when no manager is present
-type nilManager struct{}
+type nilManager struct {
+	logger *logp.Logger
+	lock   sync.Mutex
+	status Status
+	msg    string
+}
 
-func nilFactory(*common.Config, *reload.Registry, uuid.UUID) (ConfigManager, error) {
-	return nilManager{}, nil
+func nilFactory(*common.Config, *reload.Registry, uuid.UUID) (Manager, error) {
+	log := logp.NewLogger("mgmt")
+	return &nilManager{
+		logger: log,
+		status: Unknown,
+		msg:    "",
+	}, nil
 }
 
-func (nilManager) Enabled() bool                           { return false }
-func (nilManager) Start(_ func())                          {}
-func (nilManager) Stop()                                   {}
-func (nilManager) CheckRawConfig(cfg *common.Config) error { return nil }
+func (*nilManager) Enabled() bool                           { return false }
+func (*nilManager) Start(_ func())                          {}
+func (*nilManager) Stop()                                   {}
+func (*nilManager) CheckRawConfig(cfg *common.Config) error { return nil }
+func (n *nilManager) UpdateStatus(status Status, msg string) {
+	n.lock.Lock()
```

Comment:
do we need this here if this is nil manager - noop manager?

Reply:
I do it only for logging the status changes. I think it is good to show status updates in the logger as well, so it's clear what is happening in the beat, even if its not controlled by Elastic Agent.